common-utils/src/main/java/eu/arrowhead/common/intf/properties/validators/PortValidator.java

```diff
@@ -0,0 +1,46 @@
+package eu.arrowhead.common.intf.properties.validators;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import eu.arrowhead.common.Constants;
+import eu.arrowhead.common.exception.InvalidParameterException;
+import eu.arrowhead.common.intf.properties.IPropertyValidator;
+
+@Service
+public class PortValidator implements IPropertyValidator {
+
+	//=================================================================================================
+	// members
+
+	@SuppressWarnings("checkstyle:nowhitespaceafter")
+	private static String[] minMax = { String.valueOf(Constants.MIN_PORT), String.valueOf(Constants.MAX_PORT) };
+
+	private final Logger logger = LogManager.getLogger(getClass());
+
+	@Autowired
+	private MinMaxValidator minMaxValidator;
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	// propertyValue should be an integer number
+	@Override
+	public Object validateAndNormalize(final Object propertyValue, final String... args) throws InvalidParameterException {
+		logger.debug("PortValidator.validateAndNormalize started...");
+
+		if (propertyValue instanceof Number) {
```

Comment:
If you use `instanceof Long` instead of `instanceof Number`, you don't need the isReal check. (The error message is the same if the value is not `Number` or not real.)

Reply:
What if the propertyValue is an Integer or Short or Byte?

Reply:
fixed